source/ota_over_mqtt_demo.c

```diff
@@ -337,6 +393,14 @@ static SemaphoreHandle_t xBufferSemaphore;
  */
 extern MQTTAgentContext_t xGlobalMqttAgentContext;
 
+/**
+ * @brief The global array of subscription elements.
+ *
+ * @note No thread safety is required to this array, since the updates the array
+ * elements are done only from one task at a time.
+ */
+extern SubscriptionElement_t xGlobalSubscriptionList[ SUBSCRIPTION_MANAGER_MAX_SUBSCRIPTIONS ];
```

Comment:
This isn't strictly necessary, since it's already accessible as `( SubscriptionElement_t * ) xGlobalMqttAgentContext.pIncomingCallbackContext`

Reply:
That is a good point. Yea that will make it strictly tied to the MQTT agent being used. I like that idea.